packages/plugins/mocha-tests/src/lib/index.js

```diff
@@ -83,82 +106,79 @@ class MochaTestRunner {
       });
     };
 
-    async.waterfall([
-      (next) => { // request provider
-        events.request("blockchain:client:provider", "ethereum", next);
-      },
-      (provider, next) => { // set provider
-        web3 = new Web3(provider);
-        next();
-      },
-      (next) => { // get accounts
-        web3.eth.getAccounts((err, accts) => {
-          if (err) {
-            return next(err);
-          }
+    const provider = await this.events.request2("tests:blockchain:start", this.options);
+    this.web3 = new Web3(provider);
+    accounts = await this.web3.eth.getAccounts();
+    await events.request2("contracts:reset");
+    let contractFiles = await events.request2("config:contractsFiles");
 
-          accounts = accts;
-          next();
-        });
-      },
-      (next) => { // reset contracts as we might have state leakage from other plugins
-        events.request("contracts:reset", next);
-      },
-      (next) => { // get contract files
-        events.request("config:contractsFiles", next);
-      },
-      (cf, next) => {
-        plugins.emitAndRunActionsForEvent('tests:contracts:compile:before', cf, next);
+    async.waterfall([
+      (next) => {
+        this.plugins.emitAndRunActionsForEvent('tests:contracts:compile:before', contractFiles, next);
       },
-      (cf, next) => { // compile contracts
-        events.request("compiler:contracts:compile", cf, next);
+      (_contractFiles, next) => {
+        contractFiles = _contractFiles;
+        events.request("compiler:contracts:compile", _contractFiles, next);
```

Comment:
Totally outside the scope of this PR, but I'm thinking that this could be part of the test-runner itself, as it's common to most runners.